library/psa_crypto_cipher.c

```diff
@@ -265,10 +265,11 @@ psa_status_t mbedtls_psa_cipher_set_iv(
     mbedtls_psa_cipher_operation_t *operation,
     const uint8_t *iv, size_t iv_length)
 {
-    if (iv_length != operation->iv_length) {
-        return PSA_ERROR_INVALID_ARGUMENT;
+    if (operation->ctx.cipher.cipher_info->type != (MBEDTLS_CIPHER_CHACHA20 || MBEDTLS_CIPHER_CHACHA20_POLY1305)) {
```

Comment:
This looks a bit odd to (especially since we're not commenting on why ChaCha/ChaChaPoly are exempted from the requirement). It looks to me like the old check just 'accidentally' worked since none of the implemented algorithms allowed non-default IV lengths. It looks like `mbedtls_cipher_set_iv()` already performs checking, so IMO it makes sense to remove the check here instead.